File: app/src/main/java/amrabed/android/release/evaluation/data/tables/DayTable.java
package amrabed.android.release.evaluation.data.tables;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

import amrabed.android.release.evaluation.data.entities.Day;

@Dao
public interface DayTable {
    @Insert(onConflict = OnConflictStrategy.IGNORE)
    void insert(Day... days);

    @Update
    void updateDay(Day day);

    @Query("SELECT * FROM days WHERE date = :date")
    LiveData<Day> get(long date);

    @Query("SELECT * FROM days ORDER BY date ASC")
    LiveData<List<Day>> getAll();
}

File: app/src/main/java/amrabed/android/release/evaluation/data/tables/DayTable.kt
package amrabed.android.release.evaluation.data.tables

import amrabed.android.release.evaluation.data.entities.Day
import androidx.lifecycle.LiveData
import androidx.room.*

@Dao
interface DayTable {
    @Insert(onConflict = OnConflictStrategy.IGNORE)
    fun insert(vararg days: Day?)

    @Update
    fun updateDay(day: Day?)

    @Query("SELECT * FROM days WHERE date = :date")
    operator fun get(date: Long): LiveData<Day?>

    @get:Query("SELECT * FROM days ORDER BY date ASC")
    val all: LiveData<List<Day>>
}